plugins/convention/src/test/java/org/apache/struts2/convention/PackageBasedActionConfigBuilderTest.java

```diff
@@ -539,37 +578,50 @@ public Container getContainer() {
         /* org.apache.struts2.convention.actions.allowedmethods class level */
         pkgConfig = configuration.getPackageConfig("org.apache.struts2.convention.actions.allowedmethods#struts-default#/allowedmethods");
         assertNotNull(pkgConfig);
+        checkSmiValue(pkgConfig, strutsDefault, isSmiInheritanceEnabled);
         assertEquals(2, pkgConfig.getActionConfigs().size());
         verifyActionConfig(pkgConfig, "class-level-allowed-methods", ClassLevelAllowedMethodsAction.class, "execute", pkgConfig.getName());
         assertEquals("struts-default", pkgConfig.getParents().get(0).getName());
 
         ActionConfig actionConfig = pkgConfig.getActionConfigs().get("class-level-allowed-methods");
-        assertEquals(7, actionConfig.getAllowedMethods().size());
         assertTrue(actionConfig.getAllowedMethods().contains("execute"));
-        assertTrue(actionConfig.getAllowedMethods().contains("end"));
-        assertTrue(actionConfig.getAllowedMethods().contains("input"));
-        assertTrue(actionConfig.getAllowedMethods().contains("cancel"));
-        assertTrue(actionConfig.getAllowedMethods().contains("start"));
-        assertTrue(actionConfig.getAllowedMethods().contains("home"));
-        assertTrue(actionConfig.getAllowedMethods().contains("browse"));
+        if(isSmiInheritanceEnabled) {
+            // With strict method invocation disabled the allowed methods are "execute" and the wildcard "*"
+            assertEquals(2, actionConfig.getAllowedMethods().size());
```

Comment:
Maybe you could extract `actionConfig.getAllowedMethods().size()` and put it before the if stmt?